pkg/runctl/run_manager.go

```diff
@@ -342,27 +350,85 @@ func (c *runManager) setupNetworkPolicyFromConfig(ctx *runContext) error {
 		Kind:  "NetworkPolicy",
 	}
 
-	policyStr := c.pipelineRunsConfig.NetworkPolicy
-	if policyStr == "" {
+	configStr := c.pipelineRunsConfig.NetworkPolicy
+
+	if configStr == "" {
 		return nil
 	}
 
-	var obj *unstructured.Unstructured
+	c.createResource(configStr, "networkpolicies", expectedGroupKind, ctx)
+
+	return nil
+}
+
+func (c *runManager) setupLimitRanger(ctx *runContext) error {
+	if err := c.setupLimitRangerFromConfig(ctx); err != nil {
+		return errors.Wrapf(err,
+			"failed to set up the configured LimitRanger in namespace %q",
+			ctx.runNamespace,
+		)
+	}
+	return nil
+}
+
+func (c *runManager) setupLimitRangerFromConfig(ctx *runContext) error {
+	expectedGroupKind := schema.GroupKind{
+		Group: "",
+		Kind:  "LimitRange",
+	}
+
+	configStr := c.pipelineRunsConfig.LimitRange
+	if configStr == "" {
+		return nil
+	}
+
+	c.createResource(configStr, "limitranges", expectedGroupKind, ctx)
+
+	return nil
+}
+
+func (c *runManager) setupResourceQuota(ctx *runContext) error {
+	if err := c.setupResourceQuotaFromConfig(ctx); err != nil {
+		return errors.Wrapf(err,
+			"failed to set up the configured ResourceQuota in namespace %q",
+			ctx.runNamespace,
+		)
+	}
+	return nil
+}
 
+func (c *runManager) setupResourceQuotaFromConfig(ctx *runContext) error {
+	expectedGroupKind := schema.GroupKind{
+		Group: "",
+		Kind:  "ResourceQuota",
+	}
+
+	configStr := c.pipelineRunsConfig.ResourceQuota
+	if configStr == "" {
+		return nil
+	}
+
+	c.createResource(configStr, "resourcequotas", expectedGroupKind, ctx)
+
+	return nil
+}
+
+func (c *runManager) createResource(configStr string, resourceName string, expectedGroupKind schema.GroupKind, ctx *runContext) error {
```

Comment:
`resourceName` is misleading (could be the name of the API object). I'd prefer to keep the terminology of `k8s.io/apimachinery/pkg/runtime/schema`:
```suggestion
func (c *runManager) createResource(configStr string, resource string, expectedGroupKind schema.GroupKind, ctx *runContext) error {
```
For human-readable messages there should be an additional `resourceDisplayName` taking for instance `network policy` as value.